test/IntegrationTests/Microsoft.Testing.Platform.Acceptance.IntegrationTests/ExecutionTests.cs

```diff
@@ -199,22 +201,34 @@ public Task<CloseTestSessionResult> CloseTestSessionAsync(CloseTestSessionContex
 
     public async Task ExecuteRequestAsync(ExecuteRequestContext context)
     {
-        await context.MessageBus.PublishAsync(this, new TestNodeUpdateMessage(context.Request.Session.SessionUid,
-            new TestNode() { Uid = "0", DisplayName = "Test1", Properties = new(DiscoveredTestNodeStateProperty.CachedInstance) }));
+        bool isDiscovery = _sp.GetCommandLineOptions().TryGetOptionArgumentList("--list-tests", out _);
```

Comment:
As this changing test "input", I assume that now there is some behavior difference. It would be good to explicitly describe in PR description, and how is it going to affect known frameworks today (if affects them at all)

Reply:
From private chat, we now handle both events in the same handler, so if test framework previously produced both events (especially on run) the new code will break the framework because it expects discovery to be reported via discovery, and execution events to be reported  during execution but not both.

Reply:
It seems correct to me that discover events are produced during the run. I am tempted to say discovery should produce only discovery events but maybe it would be fine to allow skipped (if you know upfront this test cannot be run) or errors if there is a misconfiguration.

Maybe what we are missing is to have the output informed about the kind of request we are into (maybe just detecting if `--list-tests` is set is enough) to know what kind of info to display.

Reply:
> It seems correct to me that discover events are produced during the run.

@nohwnd Does the behavior change for this case? Let's ignore the case where non-discovery events are produced during discovery. It would be good to still test the case for producing discovery + non-discovery during run.

Reply:
handled in code, and tests fail if we revert it the if, so also tested.